Add tests wk 1

EmilePW opened the pull request:
Added some tests in the same vein as from the JS Core 1 homework.These give a bit more feedback about what might have gone wrong if the code doesn't work and tests edge cases. I'm hoping it will make the homework quicker to mark as well.

EmilePW committed:
Add tests for JS Core 2, Week 1 homework

Added some tests in the same vein as from the JS Core 1 homework.These give a bit more feedback about what might have gone wrong if the code doesn't work and tests edge cases. I'm hoping it will make the homework quicker to mark as well.

EmilePW committed:
Revert trailing commas to original formatting

Had added this after running Prettier in VS Code.

EmilePW commented:
@ChrisOwen101 - sorry, forgot to actually add a notification for this yesterday.

ChrisOwen101 commented on Week-1/Homework/mandatory/1-writers.js:
```diff
@@ -1,3 +1,14 @@
+/* ======= TESTS - DO NOT MODIFY =====
```
Could this be extracted to a separate JS file so we don't have to include it at the top of the file where it's quite invasive?

ChrisOwen101 commented on Week-1/Homework/mandatory/3-groceries.js:
```diff
@@ -1,3 +1,15 @@
+/* ======= TESTS - DO NOT MODIFY =====
```
Same as the above - could this be moved into a JS file that can be included here? Also reduced duplicated code

ChrisOwen101 commented on Week-1/Homework/mandatory/1-writers.js:
```diff
@@ -36,9 +47,59 @@ let writers = [
+To run these tests type `node 1-writers.js` into your terminal or run in VS code (F5)
+*/
+
+function test(test_name, expr, expected, actual) {
```
Maybe this function could be pulled out to another JS file as well?

ChrisOwen101 reviewed (approved):
Great work! A few small comments that would help tidy this up

EmilePW commented:
Thanks for reviewing @ChrisOwen101. I was initially thinking it might cause problems if I start using `require` depending on how they are running the code but I think it's fine as long as they've followed the CYF instructions on setting things up. I'll make the changes you suggested above later.

EmilePW committed:
Move tests into separate module

The code is cleaner now, has more explanation and shouldn't get in the student's way.

EmilePW commented:
Additional changes made above @ChrisOwen101
